### PR TITLE
feat(VILLB-18): add exceptions filter, BREAKING CHANGE: change http e…

### PR DESCRIPTION
**Pull request by katafalque**
…xceptions in services to custom errors.

### CODE REVIEW

File: src/exception/invalid-credentials.error.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export class InvalidCredentialsError extends Error {
```
Comment (ismetkizgin) [inline src/exception/invalid-credentials.error.ts:1]: Can we make the Expection nomenclature and the class from which it is inherited as in the nestjs document? 

https://docs.nestjs.com/exception-filters#custom-exceptions

File: src/exception/invalid-otp.error.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export class InvalidOtpError extends Error {
```
Comment (ismetkizgin) [inline src/exception/invalid-otp.error.ts:1]: Can we make the Expection nomenclature and the class from which it is inherited as in the nestjs document? 

https://docs.nestjs.com/exception-filters#custom-exceptions

File: src/exception/invalid-refresh-token.error.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export class InvalidRefreshTokenError extends Error {
```
Comment (ismetkizgin) [inline src/exception/invalid-refresh-token.error.ts:1]: Can we make the Expection nomenclature and the class from which it is inherited as in the nestjs document? 

https://docs.nestjs.com/exception-filters#custom-exceptions

File: src/exception/invalid-reset-password-request.error.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export class InvalidResetPasswordRequestError extends Error {
```
Comment (ismetkizgin) [inline src/exception/invalid-reset-password-request.error.ts:1]: Can we make the Expection nomenclature and the class from which it is inherited as in the nestjs document? 

https://docs.nestjs.com/exception-filters#custom-exceptions

File: src/exception/invalid-user.error.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export class InvalidUserError extends Error {
```
Comment (ismetkizgin) [inline src/exception/invalid-user.error.ts:1]: Can we make the Expection nomenclature and the class from which it is inherited as in the nestjs document? 

https://docs.nestjs.com/exception-filters#custom-exceptions

File: src/exception/user-exists.error.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export class UserExistsError extends Error {
```
Comment (ismetkizgin) [inline src/exception/user-exists.error.ts:1]: Can we make the Expection nomenclature and the class from which it is inherited as in the nestjs document? 

https://docs.nestjs.com/exception-filters#custom-exceptions

File: src/exception/user-not-found.error.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export class UserNotFoundError extends Error {
```
Comment (ismetkizgin) [inline src/exception/user-not-found.error.ts:1]: Can we make the Expection nomenclature and the class from which it is inherited as in the nestjs document? 

https://docs.nestjs.com/exception-filters#custom-exceptions

File: src/filter/exceptions.filter.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+import { Response } from "express";
+import { InvalidCredentialsError } from "../exception/invalid-credentials.error";
+import { UserExistsError } from "../exception/user-exists.error";
+import { InvalidResetPasswordRequestError } from "../exception/invalid-reset-password-request.error";
+import { InvalidRefreshTokenError } from "../exception/invalid-refresh-token.error";
+import { InvalidOtpError } from "../exception/invalid-otp.error";
+import { UserNotFoundError } from "../exception/user-not-found.error";
+
+@Catch(Error)
+export class ExceptionsFilter implements ExceptionFilter {
```
Comment (ismetkizgin) [inline src/filter/exceptions.filter.ts:20]: Expection classes will have status information. Can we review the example and update it according to the incoming data? 

https://docs.nestjs.com/exception-filters#exception-filters-1

File: src/refresh-token/refresh-token.controller.ts [hunk 9]
```diff
@@ -10,6 +11,7 @@ export class RefreshTokenController {
   ) {}
 
   @Post()
+  @UseFilters(new ExceptionsFilter())
```
Comment (ismetkizgin) [inline src/refresh-token/refresh-token.controller.ts:14]: Let's not use the Expection filter property in this way. We can define it globally only once in Main. 

`app.useGlobalFilters(new HttpExceptionFilter());`

**Review by ismetkizgin — CHANGES_REQUESTED** [review]

**Comment by mfozmen** [comment]
It will decline after merging this: https://github.com/BrewInteractive/authentication-service-nestjs/pull/99

**Comment by mfozmen** [comment]
will merge this https://github.com/BrewInteractive/authentication-service-nestjs/pull/99 instead of this pr.

**Pull request closed without merging**